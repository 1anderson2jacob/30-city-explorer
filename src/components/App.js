import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
// import superagent from 'superagent';
import Home from './Home';
import Location from './Location';


class App extends Component {

  // async componentDidMount() {
  //   let data = await superagent.get('/location');
  //   console.log(data, 'data yay!');
  // }


  render() {
    return (
      <Router>
        <div>
          <Route exact path="/" component={Home} />
          <Route exact path="/:location" component={Location} />
        </div>
      </Router>
    );
  }
}

export default App;
