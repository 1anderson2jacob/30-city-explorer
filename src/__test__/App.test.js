import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });
import App from '../components/App';


describe('running tests on city explorer App', () => {
  it('is alive', () => {
    let component = shallow(<App />);
    expect(component.find('div').exists()).toBeTruthy();
  });
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders correctly', () => {
    let render = renderer.create('<App/>').toJSON();
    expect(render).toMatchSnapshot();
  });
});


