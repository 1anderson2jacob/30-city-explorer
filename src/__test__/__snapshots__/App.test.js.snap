// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`running tests on city explorer App renders correctly 1`] = `"<App/>"`;
